manifests/config.pp

```diff
@@ -3,78 +3,82 @@
-  $config_additional_variables = $rabbitmq::config_additional_variables
-  $auth_backends               = $rabbitmq::auth_backends
-  $cluster_partition_handling  = $rabbitmq::cluster_partition_handling
-  $file_limit                  = $rabbitmq::file_limit
-  $collect_statistics_interval = $rabbitmq::collect_statistics_interval
-  $ipv6                        = $rabbitmq::ipv6
-  $inetrc_config               = $rabbitmq::inetrc_config
-  $inetrc_config_path          = $rabbitmq::inetrc_config_path
-  $ssl_erl_dist                = $rabbitmq::ssl_erl_dist
+  $admin_enable                 = $rabbitmq::admin_enable
+  $cluster_node_type            = $rabbitmq::cluster_node_type
+  $cluster_nodes                = $rabbitmq::cluster_nodes
+  $config                       = $rabbitmq::config
+  $config_cluster               = $rabbitmq::config_cluster
+  $config_path                  = $rabbitmq::config_path
+  $config_ranch                 = $rabbitmq::config_ranch
+  $config_stomp                 = $rabbitmq::config_stomp
+  $config_shovel                = $rabbitmq::config_shovel
+  $config_shovel_statics        = $rabbitmq::config_shovel_statics
+  $default_user                 = $rabbitmq::default_user
+  $default_pass                 = $rabbitmq::default_pass
+  $env_config                   = $rabbitmq::env_config
+  $env_config_path              = $rabbitmq::env_config_path
+  $erlang_cookie                = $rabbitmq::erlang_cookie
+  $interface                    = $rabbitmq::interface
+  $management_port              = $rabbitmq::management_port
+  $management_ssl               = $rabbitmq::management_ssl
+  $management_hostname          = $rabbitmq::management_hostname
+  $node_ip_address              = $rabbitmq::node_ip_address
+  $rabbitmq_user                = $rabbitmq::rabbitmq_user
+  $rabbitmq_group               = $rabbitmq::rabbitmq_group
+  $rabbitmq_home                = $rabbitmq::rabbitmq_home
+  $port                         = $rabbitmq::port
+  $tcp_keepalive                = $rabbitmq::tcp_keepalive
+  $tcp_backlog                  = $rabbitmq::tcp_backlog
+  $tcp_sndbuf                   = $rabbitmq::tcp_sndbuf
+  $tcp_recbuf                   = $rabbitmq::tcp_recbuf
+  $heartbeat                    = $rabbitmq::heartbeat
+  $service_name                 = $rabbitmq::service_name
+  $ssl                          = $rabbitmq::ssl
+  $ssl_only                     = $rabbitmq::ssl_only
+  $ssl_cacert                   = $rabbitmq::ssl_cacert
+  $ssl_cert                     = $rabbitmq::ssl_cert
+  $ssl_key                      = $rabbitmq::ssl_key
+  $ssl_depth                    = $rabbitmq::ssl_depth
+  $ssl_cert_password            = $rabbitmq::ssl_cert_password
+  $ssl_port                     = $rabbitmq::ssl_port
+  $ssl_interface                = $rabbitmq::ssl_interface
+  $ssl_management_port          = $rabbitmq::ssl_management_port
+  $ssl_management_cacert_enable = $rabbitmq::ssl_management_cacert_enable
```

Comment:
Can you please add datatypes for the new params?

Reply:
I think this is because they're not class params, but variables in `config.pp`. in `init.pp`, these all have data types. IIRC, you can't specify data types for variables vs. class params. I think they're not settable since they're in the body of the manifest, though we could also add an explicit `assert_private()` too maybe?